### PR TITLE
clarify README.md

### PR DESCRIPTION
**Pull request by wds15**
## Summary

Adapt README.md to our new logic of instructions.

## Release notes

clarify README.md

## Checklist

- [X] Math issue #1820 

- [X] Copyright holder: Sebastian Weber

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [X] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [X] the code is written in idiomatic C++ and changes are documented in the doxygen

- [X] the new changes are tested

### CODE REVIEW

File: stan/math/opencl/kernel_generator/load.hpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ class load_
    * Creates a deep copy of this expression.
    * @return copy of \c *this
    */
-  inline load_<T&> deep_copy() const& { return load_<T&>(a_); }
+  inline load_<T&> deep_copy() const & { return load_<T&>(a_); }
```
Comment (rok-cesnovar) [inline stan/math/opencl/kernel_generator/load.hpp:52]: Can you revert this change? Without this change the tests will pass without running the entire suite.

Comment (wds15) [reply stan/math/opencl/kernel_generator/load.hpp:52]: Maybe... I haven't asked for this change - the Jenkins did this for me. Let me try.

Comment (rok-cesnovar) [reply stan/math/opencl/kernel_generator/load.hpp:52]: Ok, I see why this happened (some machine has clang-format 5.0.0 and some have 6.0.0). Dont bother too much if it wont work.

**Comment by wds15** [comment]
Hopefully Jenkins will not change back this formatting thing (which pops up in all PRs right now for whatever reason).

**Comment by rok-cesnovar** [comment]
Yes some Jenkins workers have clang-format 5.0.0 and some have 6.0.0 and they differ in some details and that needs to be sorted out.
